### PR TITLE
crush redundant variables in rules and have one in the main workflow

### PR DESCRIPTION
**Pull request by hassanfa**
### This PR:

Changed:

- move redundant variables into main workflow

### Review and tests: 
- [ ] Tests pass
- [ ] Code review
- [ ] New code is executed and covered by tests, and test approve

### CODE REVIEW

**Comment by hassanfa** [comment]
Screenshots coming!

WGS T-only config successful:
image

WGS TN config successful:
image

Panel T-only config successful:
image

Panel TN config successful:
image

**Comment by hassanfa** [comment]
@ashwini06 ready to review!

File: BALSAMIC/snakemake_rules/quality_control/mosdepth.rule [hunk 1]
```diff
@@ -1,9 +1,7 @@
 # vim: syntax=python tabstop=4 expandtab
 # coding: utf-8
 
-from BALSAMIC.utils.rule import get_conda_env, get_picard_mrkdup
-
-picarddup = get_picard_mrkdup(config)
+from BALSAMIC.utils.rule import get_conda_env
```
Comment (ashwini06) [inline BALSAMIC/snakemake_rules/quality_control/mosdepth.rule:5]: Missing 
`from BALSAMIC.utils.rule import get_threads`

File: BALSAMIC/snakemake_rules/variant_calling/somatic_tumor_only.rule [hunk 2]
```diff
@@ -3,16 +3,10 @@
...
 from BALSAMIC.utils.rule import get_conda_env
 from BALSAMIC.utils.rule import get_sample_type
 from BALSAMIC.utils.rule import get_threads
 
-picarddup = get_picard_mrkdup(config)
-chromlist = config["panel"]["chrom"]
-capture_kit = os.path.split(config["panel"]["capture_kit"])[1]
-
-
 def get_pon(config):
```
Comment (ashwini06) [inline BALSAMIC/snakemake_rules/variant_calling/somatic_tumor_only.rule:10]: how about moving this def to `utils/rule.py`?

File: BALSAMIC/snakemake_rules/variant_calling/mergetype.rule [hunk 3]
```diff
@@ -4,14 +4,8 @@
 from BALSAMIC.utils.rule import get_sample_type
```
Comment (ashwini06) [inline BALSAMIC/snakemake_rules/variant_calling/mergetype.rule:4]: `get_sample_type` defined in the main workflow. Do you need it again here?

File: BALSAMIC/snakemake_rules/variant_calling/mergetype.rule [hunk 4]
```diff
@@ -4,14 +4,8 @@
...
 from BALSAMIC.utils.rule import get_conda_env, get_picard_mrkdup
```
Comment (ashwini06) [inline BALSAMIC/snakemake_rules/variant_calling/mergetype.rule:5]:
```suggestion
from BALSAMIC.utils.rule import get_conda_env
```

File: BALSAMIC/snakemake_rules/variant_calling/somatic_tumor_normal.rule [hunk 5]
```diff
@@ -7,9 +7,6 @@ from BALSAMIC.utils.rule import get_conda_env
 from BALSAMIC.utils.rule import get_sample_type
```
Comment (ashwini06) [inline BALSAMIC/snakemake_rules/variant_calling/somatic_tumor_normal.rule:7]:
```suggestion

```

File: BALSAMIC/snakemake_rules/variant_calling/somatic_tumor_only.rule [hunk 6]
```diff
@@ -3,16 +3,10 @@
 
 import os
 
-from BALSAMIC.utils.rule import get_picard_mrkdup
 from BALSAMIC.utils.rule import get_conda_env
 from BALSAMIC.utils.rule import get_sample_type
```
Comment (ashwini06) [inline BALSAMIC/snakemake_rules/variant_calling/somatic_tumor_only.rule:7]:
```suggestion
from BALSAMIC.utils.rule import get_sample_type
```

**Review by ashwini06 — COMMENTED** [review]
Looks good 👍  Just a few comments.

File: BALSAMIC/snakemake_rules/quality_control/mosdepth.rule (hunk 1 above)
Comment (hassanfa) [reply BALSAMIC/snakemake_rules/quality_control/mosdepth.rule]: fixed

File: BALSAMIC/snakemake_rules/variant_calling/mergetype.rule (hunk 3 above)
Comment (hassanfa) [reply BALSAMIC/snakemake_rules/variant_calling/mergetype.rule]: fixed

File: BALSAMIC/snakemake_rules/variant_calling/mergetype.rule (hunk 4 above)
Comment (hassanfa) [reply BALSAMIC/snakemake_rules/variant_calling/mergetype.rule]: fixed

File: BALSAMIC/snakemake_rules/variant_calling/somatic_tumor_normal.rule (hunk 5 above)
Comment (hassanfa) [reply BALSAMIC/snakemake_rules/variant_calling/somatic_tumor_normal.rule]: fixed

File: BALSAMIC/snakemake_rules/variant_calling/somatic_tumor_only.rule (hunk 6 above)
Comment (hassanfa) [reply BALSAMIC/snakemake_rules/variant_calling/somatic_tumor_only.rule]: fixed

File: BALSAMIC/snakemake_rules/variant_calling/somatic_tumor_only.rule (hunk 2 above)
Comment (hassanfa) [reply BALSAMIC/snakemake_rules/variant_calling/somatic_tumor_only.rule:4]: as discussed, refactoring this one and other functions in rules will be done in another PR.

**Comment by hassanfa** [comment]
Mergin!

**Pull request merged**